templates/zerver/app/message_feed_errors.html

```diff
@@ -15,7 +15,10 @@
         End of results from your
         <a href="/help/search-for-messages#searching-shared-history"
           target="_blank" rel="noopener noreferrer">history</a>.
-        Consider <a class="search-shared-history" href="">searching all public streams</a>.
+        <span class="only-visible-for-spectators"><a href="/login/">Log in</a> to search your personal history.</span>
+        <span class="hidden-for-spectators">
+            Consider <a class="search-shared-history" href="">searching all public streams</a>.
+        </span>
         {% endtrans %}
```

Comment:
I don't like having so much HTML in here; translators will have to retranslate this whole block every time we change any of the class names.  Can you take the opportunity to move this block into handlebars and use the `z-link` approach so that the strings for translation are just the actual strings with some blocks?  

The actual UX looks good.

Reply:
Yeah, done!